geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/serialization/AvroSerialization.scala

```diff
@@ -108,5 +109,23 @@ class AvroReader extends AbstractReader[Decoder] {
     decoder.readArrayStart.asInstanceOf[Int]
   }
 
+  // Avro specific reading method...see Decoder javadoc
+  override def readGenericMap(version: Version): DatumReader[Decoder, java.util.Map[AnyRef, AnyRef]] = (reader) => {
```

Comment:
The double loop inside makes it seem as if what should come back is a list of maps, not a single map.  Oh, well...  Guess I'll have to read the Avro docs to figure out what they're thinking.